frontend/src/metabase/visualizations/lib/table_format.js

```diff
@@ -127,17 +127,21 @@ function compileFormatter(
     }
     switch (operator) {
       case "<":
-        return v => (v < value ? color : null);
+        return v => (typeof value === "number" && v < value ? color : null);
       case "<=":
-        return v => (v <= value ? color : null);
+        return v => (typeof value === "number" && v <= value ? color : null);
       case ">=":
-        return v => (v >= value ? color : null);
+        return v => (typeof value === "number" && v >= value ? color : null);
       case ">":
-        return v => (v > value ? color : null);
+        return v => (typeof value === "number" && v > value ? color : null);
       case "=":
         return v => (v === value ? color : null);
```

Comment:
Mise well fix #8478 while you are in here. Either that or merge #8545

Reply:
I think it should already be fixed (probably only for new rules)